spec/ruby-push-notifications/wns/wns_notification_spec.rb

```diff
@@ -0,0 +1,68 @@
+module RubyPushNotifications
+  module WNS
+    describe WNSNotification do
+
+      let(:device_urls) { %w(a b c) }
+      let(:raw_notification) { build :wns_notification }
+      let(:toast_data) { { title: 'Title', message: 'Hello WNS World!', type: :toast } }
+      let(:toast_notification) { build :wns_notification, device_urls: device_urls, data: toast_data }
+      let(:tile_data) { { message: 'Hello WNS World!', image: 'http://image.com/image.jpg', type: :tile } }
+      let(:tile_notification) { build :wns_notification, device_urls: device_urls, data: tile_data }
+      let(:badge_data) { { value: 10, type: :badge } }
+      let(:badge_notification) { build :wns_notification, device_urls: device_urls, data: badge_data }
+
+      it 'builds the right wns raw xml' do
+        xml = '<?xml version="1.0" encoding="UTF-8"?>'
+        xml << '<root><value1>hello</value1></root>'
+        expect(raw_notification.as_wns_xml).to eq xml
+      end
+
+      it 'builds the right wns badge xml' do
+        xml = '<?xml version="1.0" encoding="UTF-8"?>'
+        xml << '<badge value="10"/>'
+        expect(badge_notification.as_wns_xml).to eq xml
+      end
+
+      it 'builds the right wns toast xml' do
```

Comment:
I'd like to see this test case extended a little bit to cover `launch_params` as well

Reply:
added